invenio_app_ils/records/api.py

```diff
@@ -25,56 +33,66 @@ def get_record_by_pid(cls, pid, with_deleted=False):
             object_type="rec",
             getter=cls.get_record,
         )
-        persistent_identifier, record = resolver.resolve(str(pid))
+        _, record = resolver.resolve(str(pid))
         return record
 
+    @classmethod
+    def create(cls, data, id_=None, **kwargs):
+        """Create Document record."""
+        data["$schema"] = current_jsonschemas.path_to_url(cls._schema)
+        return super(IlsRecord, cls).create(data, id_=id_, **kwargs)
+
 
 class Document(IlsRecord):
     """Document record class."""
 
-    _pid_type = 'docid'
+    pid_field = "document_pid"
+    _pid_type = DOCUMENT_PID_TYPE
     _schema = "documents/document-v1.0.0.json"
 
-    def __init__(self, data, model=None):
-        """."""
-        super(Document, self).__init__(data, model)
-
-    @classmethod
-    def create(cls, data, id_=None, **kwargs):
-        """Create Document record."""
-        data["$schema"] = current_jsonschemas.path_to_url(cls._schema)
-        return super(Document, cls).create(data, id_=id_, **kwargs)
-
 
 class Item(IlsRecord):
     """Item record class."""
 
-    _pid_type = 'itemid'
+    pid_field = "item_pid"
+    _pid_type = ITEM_PID_TYPE
     _schema = "items/item-v1.0.0.json"
 
-    def __init__(self, data, model=None):
-        """."""
-        super(Item, self).__init__(data, model)
-
     @classmethod
     def create(cls, data, id_=None, **kwargs):
         """Create Item record."""
-        data["$schema"] = current_jsonschemas.path_to_url(cls._schema)
+        data["circulation_status"] = {
+            "$ref": "{scheme}://{host}/api/circulation/items/{pid_value}/loan"
```

Comment:
Another thing, this url is hardcoded, and it is the same used by the `jsonresolver` decorator in the `resolvers.py` module. Any idea how we put them in common? Maybe we can use a constant?

Reply:
could we get the URL by url_for or similar?